ai-catalog/samples/gemini-live-todo/src/main/java/com/android/ai/samples/geminilivetodo/ui/TodoScreen.kt

```diff
@@ -0,0 +1,225 @@
+import kotlin.collections.reversed
+
+/**
+ * The main screen for the To-do list application.
+ * This composable is stateful, connecting to the ViewModel to manage UI state and events.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@RequiresPermission(Manifest.permission.RECORD_AUDIO)
+@Composable
+fun TodoScreen(viewModel: TodoScreenViewModel = hiltViewModel()) {
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+    var text by remember { mutableStateOf("") }
+
+    LaunchedEffect(Unit) {
+        viewModel.initializeGeminiLive()
+    }
+
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                colors = TopAppBarDefaults.topAppBarColors(
+                    containerColor = MaterialTheme.colorScheme.primaryContainer,
+                    titleContentColor = MaterialTheme.colorScheme.primary,
+                ),
+                title = { Text(stringResource(R.string.gemini_live_title)) },
+            )
+        },
+        floatingActionButton = {
+            if (uiState is TodoScreenUiState.Success) {
+                val successState = uiState as TodoScreenUiState.Success
+                val micIcon = when {
+                    !successState.isLiveSessionReady -> Icons.Filled.MicOff
+                    successState.isLiveSessionRunning -> Icons.Filled.Mic
+                    else -> Icons.Filled.MicNone
+                }
+
+                val containerColor = if (successState.isLiveSessionRunning) {
+                    val infiniteTransition =
+                        rememberInfiniteTransition(label = "mic_color_transition")
+                    infiniteTransition.animateColor(
+                        initialValue = MaterialTheme.colorScheme.primary.copy(alpha = 0.2f),
+                        targetValue = MaterialTheme.colorScheme.primary.copy(alpha = 0.9f),
+                        animationSpec = infiniteRepeatable(
+                            animation = tween(1000, easing = LinearEasing),
+                            repeatMode = RepeatMode.Reverse,
+                        ),
+                        label = "mic_color",
+                    ).value
+                } else {
+                    MaterialTheme.colorScheme.primaryContainer
+                }
+
+                FloatingActionButton(
+                    onClick = { if (successState.isLiveSessionReady) viewModel.toggleLiveSession() },
+                    containerColor = containerColor,
+                ) {
+                    Icon(micIcon, stringResource(R.string.interact_with_todolist_by_voice))
+                }
+            }
+        },
+        floatingActionButtonPosition = FabPosition.Center,
+        modifier = Modifier.fillMaxSize(),
+    ) { paddingValues ->
+        Column(
+            modifier = Modifier
+                .padding(paddingValues)
+                .padding(16.dp)
+                .imePadding()
+                .fillMaxSize(),
+        ) {
+            Row(
```

Comment:
Please split into individual composables for input and content, also consider adding previews